auth-foundation/src/main/java/com/okta/authfoundation/client/internal/NetworkUtils.kt

```diff
@@ -155,6 +156,8 @@ private suspend fun OidcConfiguration.executeRequest(
             }.build()
             response = okHttpClient.newCall(newRequest).await()
             if (response.code != rateLimitErrorCode) break
+
+            event = RateLimitExceededEvent(request, response)
```

Comment:
We need to send this event every time.

That being said, we should probably refactor this to be a do/while instead of dropping into this when we get a 429 (happy to discuss offline).

We also need to add tests to ensure we send the correct amount of events.

Reply:
I refactored to do/while, and have added tests for number of events as well